src/main/java/net/minecraftforge/fmp/microblock/MicroblockContainer.java

```diff
@@ -0,0 +1,135 @@
+package net.minecraftforge.fmp.microblock;
+
+import java.util.Collection;
+import java.util.UUID;
+
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.IMultipartContainer;
+import net.minecraftforge.fmp.multipart.MultipartContainer;
+import net.minecraftforge.fmp.multipart.PartSlot;
+
+/**
+ * Helper class that wraps a {@link MultipartContainer} so that only microblocks can be added to it and forwards all the
+ * other methods.
+ *
+ * @see IMicroblock
+ * @see IMultipartContainer
+ * @see MultipartContainer
+ */
+public class MicroblockContainer implements IMultipartContainer
+{
+    
+    private IMicroblockContainerTile microTile;
+    private MultipartContainer container;
+
+    public MicroblockContainer(IMicroblockContainerTile microTile)
+    {
+        this.microTile = microTile;
+        this.container = new MultipartContainer(microTile, false);
+    }
+
+    public MultipartContainer getPartContainer()
+    {
+        return container;
+    }
+
+    @Override
+    public World getWorldIn()
+    {
+        return container.getWorldIn();
+    }
+
+    @Override
+    public BlockPos getPosIn()
+    {
+        return container.getPosIn();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Collection<? extends IMicroblock> getParts()
+    {
```

Comment:
Since MicroblockContainer implements IMultipartContainer and that one returns Collection<? extends IMultipart> I can just do Collection<? extends IMicroblock> so people can iterate through them directly knowing that they are microblocks, without having to cast every single one manually.